docs/features/software-catalog/external-integrations.md

```diff
@@ -159,7 +159,12 @@ Check out the numbered markings - let's go through them one by one.
    the outcome of that. This example issues a `fetch` to the right service and
    issues a full refresh of its entity bucket based on that.
 5. The method translates the foreign data model to the native `Entity` form, as
-   expected by the catalog.
+   expected by the catalog. Make sure that in this method you include the
+   `backstage.io/managed-by-location` and `backstage.io/managed-by-origin-location`
+   annotations on your `Entity`. If these are not present they will not show up
+   in the Catalog and you will see warnings in your logs. The
+   [Well-known Annotations](./well-known-annotations.md#backstageiomanaged-by-location)
+   documentation has guidance on what values to use for these.
```

Comment:
How about:
> The `Entity` must include the `backstage.io/managed-by-location` and `backstage.io/managed-by-origin-location` annotations; otherwise, it will not appear in the Catalog and will generate warning logs. The [Well-known Annotations](./well-known-annotations.md#backstageiomanaged-by-location) documentation has guidance on what values to use for these.

I've made this suggestion for a few reasons:
1. The purpose of the "method translates..." is about creating an Entity. My suggested change keeps the focus on the Entity and what it needs rather than what the method should do.
2. "If these are not present they will not show up...". This statement contained multiple generic plurals (these and they) but each is referring to something different. Hopefully my proposal removes the ambiguity.

The [Example User Entity Provider](https://backstage.io/docs/features/software-catalog/external-integrations#example-user-entity-provider) section includes a detailed example of translating the data into an `Entity`. The example even includes the annotations. This raised a couple of thoughts for me:
1. Maybe the section you modified is meant to be high-level while the example is detailed. If so, does it make sense to add the note about the annotations at the high-level?
2. If the answer to 1 is yes, should the annotations be explicitly named? Or should it somehow use the `ANNOTATION_LOCATION` like the example?
3. If the answer to 1 is no, then should the explanation you provided be moved to the example instead?

Reply:
Awesome, your wording is much better so I'm just going to use that. Many thanks @GoFightNguyen 

Regarding the other items you mentioned:

1. Yes, high-level or not without the information about the annotations your provider does not function. 
2. Yes, the exact annotations should be named as they will help you when looking at the Well-known Annotation documentation to determine what value you should be using.
3. Moving this down to the example is "burying the lede". There is a good chance you have written your provider, installed it, and hit the warnings by this point. We've seen this with the person you helped on Discord, I'm pretty confident that I've seen others do this as well on Discord, and I know I personally ran into this issue as well. 

That being said, it might be worth adding a comment in the example code about this to cover those who might jump right to it without reading the rest of the documentation. I'm just not sure exactly what it should say?